massa-factory-worker/src/endorsement_factory.rs

```diff
@@ -62,7 +62,7 @@ impl EndorsementFactoryWorker {
         // get delayed time
         let shifted_now = MassaTime::now(self.cfg.clock_compensation_millis)
             .expect("could not get current time")
-            .saturating_sub(self.half_t0);
+            .saturating_sub(self.cfg.t0);
 
         // if it's the first computed slot, add a time shift to prevent double-production on node restart with clock skew
         let base_time = if previous_slot.is_none() {
```

Comment:
I'm very unsure about that changes :roll_eyes: I miss the logic of that timing, if we can add a documentation and an example it would be great!

Reply:
NB. that's not the unique change in that file obviously

Reply:
I made a proposal here: https://github.com/massalabs/massa/pull/3004/commits/0d6d08178019fcddbc6c16adb147de101bef3e21